c10/cuda/CUDACachingAllocator.h

```diff
@@ -151,18 +151,23 @@ struct TraceEntry {
       Action action,
       int64_t addr,
       size_t size,
+      size_t block_size,
       cudaStream_t stream,
       std::shared_ptr<Context> context = nullptr)
       : action_(action),
         addr_(addr),
         context_(context),
         stream_(stream),
-        size_(size) {}
+        size_(size),
+        block_size_(block_size) {}
```

Comment:
Instead of storing block_size, let's store the amount of amount of memory rounded away `rounded_by = (block_size_ - size_)`. This is more general (it can be adapted to any allocator where an action introduces waste), and it gives meaning to some events like SEGEMENT_ALLOC which don't round off anything at all (and have the value 0). In the future if we encode the events differently to save space this will also give much smaller numbers which will compress better.

Reply:
Resolved by adding `rounded_by` along `size` in `TraceEntry` struct